components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/RequestObjectValidatorImpl.java

```diff
@@ -280,6 +319,27 @@ protected boolean isSignatureVerified(SignedJWT signedJWT, Certificate x509Certi
         }
     }
 
+    /**
+     * Validate the signedJWT signature when jwks end point is given instead of public certificate.
+     *
+     * @param jwtString
+     * @param jwksUri
+     * @param algorithm
+     * @param options
+     * @return
+     */
+    protected boolean isSignatureVerified(String jwtString, String jwksUri, String algorithm, Map<String, Object> options) throws OAuthClientAuthnException {
+
+        try {
+            return  new JWKSBasedJWTValidator().validateSignature(jwtString, jwksUri, algorithm, options);
+        } catch (IdentityOAuth2Exception e) {
+            String errorMessage = "Error occurred while validating signature using jwks";
+            throw new OAuthClientAuthnException(errorMessage, OAuth2ErrorCodes.INVALID_REQUEST, e);
```

Comment:
Is this correct to reuse this OAuthClientAuthnException here ?. AFAIK this is used to convey an oauth client authentication failure. Ex : Using invalid clientid : secret or using invalid certs by a client to authenticate to token endpoint

Reply:
changed the exception